src/index.ts

```diff
@@ -1,45 +1,185 @@
-function _(selector: string): any {
-	/**
-	 * innerHTML() {
-	 * }
-	 *
-	 * show() {
-	 * }
-	 *
-	 * hidden() {
-	 * }
-	 *
-	 * addEvent() {
-	 * }
-	 */
+function _(selector: string): HTMLElement | null {
+  const element = document.querySelector<HTMLElement>(selector);
+  if (element === null) throw new Error("존재하지 않는 element입니다.");
+
+  function setInnerHTML(str?: string) {
+    if (element === null) throw new Error("존재하지 않는 element입니다.");
+    element.innerHTML = str || "";
+  }
+
+  function showElement() {
+    if (element === null) throw new Error("존재하지 않는 element입니다.");
+    element.style.display = "block";
+  }
+
+  function hideElement() {
+    if (element === null) throw new Error("존재하지 않는 element입니다.");
+    element.style.display = "hidden";
+  }
+
+  function addEvent<T extends keyof HTMLElementEventMap>(
+    type: T,
+    listener: (event: HTMLElementEventMap[T]) => void
+  ) {
+    if (element === null) throw new Error("존재하지 않는 element입니다.");
+    element.addEventListener(type, listener);
+  }
+
+  element.setInnerHTML = setInnerHTML;
+  element.showElement = showElement;
+  element.hideElement = hideElement;
+  element.addEvent = addEvent;
+
+  return element;
+}
+declare global {
+  interface HTMLElement {
+    setInnerHTML(str?: string): void;
+    showElement(): void;
+    hideElement(): void;
+    addEvent<T extends keyof HTMLElementEventMap>(
+      type: T,
+      listener: (event: HTMLElementEventMap[T]) => void
+    ): void;
+  }
 }
 
 module _ {
-	export function fetch() {
-		return {};
-	}
+  export function fetch(
+    url: string,
+    responseType: XMLHttpRequestResponseType = "text"
+  ): Promise<string> {
+    return new Promise((resolve, reject) => {
+      try {
+        const request = new XMLHttpRequest();
+
+        request.addEventListener("load", () => {
+          if (responseType === "text") {
+            try {
+              resolve(JSON.parse(request.responseText));
+            } catch (_) {
+              resolve(request.responseText);
+            }
+          } else {
+            resolve(request.response);
+          }
+        });
+        request.addEventListener("error", () => {
+          throw new Error(String(request.status));
+        });
+        request.open("GET", url, true);
+        request.responseType = responseType;
+        request.send();
+      } catch (error) {
+        reject(error);
+      }
+    });
+  }
+
+  export function isNull<T extends unknown>(value: T): boolean {
+    return value === null;
+  }
```

Comment:
```suggestion
  export function isNull(value: unknown): boolean {
    return value === null;
  }
```

민초 궁금한게 있는데 이렇게 제네릭을 사용하지 않은 것과는 차이가 있나요?

Reply:
https://stackoverflow.com/questions/71502196/what-difference-between-extends-and-in-generic-types

Reply:
글쎄요 🤣 현재 상황에서는 뭐가 다른지 저도 잘 모르겠어요 ㅠ 제네릭이 좀 더 간지나긴하네용

Reply:
ㅋㅋㅋ간지는 제네릭이 짱이긴하죠. 이게 진짜 ts다!